src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2556,6 +2556,37 @@ theorem not_extractLsb'_append_not_extractLsb'_eq_not_extractLsb' {x : BitVec w}
     congr 1
     omega
 
+/-- A sign extension of `x : BitVec w`
+equals high bits of either `0` or `1` depending on `x.msb`,
+followed by the low bits of `x`. -/
+theorem signExtend_eq_append_extractLsb' {w v : Nat} {x : BitVec w} :
+    x.signExtend v =
+    ((if x.msb then allOnes (v - w) else 0#(v - w)) ++ x.extractLsb' 0 (min w v)).cast (by omega) := by
+  ext i hi
+  simp only [getElem_cast]
+  by_cases hx : x.msb
```

Comment:
```suggestion
  cases hx : x.msb
```
There should be no need for a by_cases here

Reply:
Will be done in the next commit.